main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripAndShapeDistanceValidator.java

```diff
@@ -32,55 +35,75 @@
  */
 @GtfsValidator
 public class TripAndShapeDistanceValidator extends FileValidator {
-
   private final GtfsTripTableContainer tripTable;
-
   private final GtfsStopTimeTableContainer stopTimeTable;
-
+  private final GtfsStopTableContainer stopTable;
   private final GtfsShapeTableContainer shapeTable;
+  private final double DISTANCE_THRESHOLD = 11.1; // distance in meters
 
   @Inject
   TripAndShapeDistanceValidator(
       GtfsTripTableContainer tripTable,
       GtfsStopTimeTableContainer stopTimeTable,
+      GtfsStopTableContainer stopTable,
       GtfsShapeTableContainer shapeTable) {
     this.tripTable = tripTable;
     this.stopTimeTable = stopTimeTable;
     this.shapeTable = shapeTable;
+    this.stopTable = stopTable;
   }
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
-    shapeTable
-        .byShapeIdMap()
+    tripTable
+        .getEntities()
         .forEach(
-            (shapeId, shape) -> {
-              double maxShapeDist =
+            trip -> {
+              String shapeId = trip.shapeId();
+              // Get distance for trip
+              int nbStopTimes = stopTimeTable.byTripId(trip.tripId()).size();
+              if (nbStopTimes == 0) {
+                return;
+              }
+              GtfsStopTime lastStopTime =
+                  stopTimeTable.byTripId(trip.tripId()).get(nbStopTimes - 1);
```

Comment:
There's a bit of a waste of time calling byTripId(trip.tripId()) here and above. Don't know if it's significant.

Reply:
done ✅